lib/binding_rust/lib.rs

```diff
@@ -603,7 +624,18 @@ impl Parser {
     /// If a pointer is assigned, then the parser will periodically read from
     /// this pointer during parsing. If it reads a non-zero value, it will halt early,
     /// returning `None`. See [parse](Parser::parse) for more information.
-    pub unsafe fn set_cancellation_flag(&self, flag: Option<&AtomicUsize>) {
+    ///
+    /// # Safety
+    ///
+    /// This can only be called when
+    /// - There is a guaranty that the flag wouldn't be dropped by the end of a scope.
+    ///
+    pub unsafe fn set_cancellation_flag_unchecked(&mut self, flag: Option<&AtomicUsize>) {
+        self.1 = None;
+        self.ts_set_cancellation_flag(flag);
+    }
+
+    unsafe fn ts_set_cancellation_flag(&self, flag: Option<&AtomicUsize>) {
```

Comment:
What do you think about combining this `ts_set_cancellation_flag` method with the public unsafe version, `set_cancellation_flag_unchecked`? I think that the safe method, `set_cancellation_flag`, could just call `set_cancellation_flag_unchecked` and *then* assign the `self.1` field to the arc. It's pretty minor, but I think it would make things slightly more clear to me.

Reply:
Good point, I did. The main idea is to prohibit to mix safe and unsafe api calls to set the flag, but this it seems wasn't the good implementation.